jujugui/static/gui/src/app/components/deployment-bar/deployment-bar.js

```diff
@@ -22,10 +22,16 @@ YUI.add('deployment-bar', function() {
 
   juju.components.DeploymentBar = React.createClass({
     propTypes: {
+      changeState: React.PropTypes.func.isRequired,
+      currentChangeSet: React.PropTypes.object.isRequired,
```

Comment:
I'm wondering if you shouldn't instead pass a 'getCurrentChangeSet' method and then everywhere you'd use this, you'd fetch the existing changeset. This would have the side effect of also being useful in the `_updateHasCommits` which would simply become a length check on the return value of this method.